src/firmware/shared/circular_buffer.h

```diff
@@ -0,0 +1,72 @@
+#include <stdbool.h>
+#include <stdlib.h>
+
+/**
+ * This struct represents the circular buffer properties
+ */
+typedef struct CircularBuffer CircularBuffer_t;
+
+/**
+ * Initialize the circular_buffer struct members
+ * @param  size Size of the circular_buffer to set
+ *
+ * @pre: `size` must be larger than 0
+ *
+ * @return A pointer to the created circular_buffer, ownership is given to the caller
+ */
+CircularBuffer_t *circular_buffer_create(size_t size);
+
+/**
+ * Destroy the circular_buffer, freeing any memory allocated for it
+ *
+ * @param  cbuffer The circular_buffer to destroy
+ */
+void circular_buffer_destroy(CircularBuffer_t *cbuffer);
+
+/**
+ * Push data into the circular_buffer
+ * If full, will overwrite the oldest value
+ * @param  cbuffer The circular_buffer
+ * @param  data  Value to be inserted into the circular_buffer
+ */
```

Comment:
Another big nit:

You have whitespace between description and @params in the other headers. Maybe make this one consistent.

Reply:
Done, made sure whitespace exists between description and params for all.